other/majority_vote_algorithm.py

```diff
@@ -0,0 +1,36 @@
+import collections
+
+"""
+This is Booyer-Moore Majority Vote Algorithm. The problem statement goes like:
+Given an integer array of size n, find all elements that appear more than ⌊ n/k ⌋ times.
+We have to solve in O(n) time and O(1) Space.
+URL : https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_majority_vote_algorithm
+"""
+
+
+def majority_element(total_votes: list[int], min_votes_required: int) -> list[int]:
```

Comment:
If `partition` is `min_votes_required` then how does this algorithm use a _constant number of words of memory_ as discussed in the Wikipedia article?  If there are 30 candidates then there will ~10 times as much memory used than 3 candidates.

Reply:
Problem : We need to find the elements (votes or candidates) that appear at least  floor( total votes / k)  times ,( here k is partitioning the array i.e. total_votes,  that's why I named it partition before ). We need to solve in a linear time. This is not representing real world voting. 


This was a problem came in an interview that I attended.  Assuming that we have all the votes and the factor is fixed like 2 or 3. What I have contributed is a general case of the algorithm.

Reply:
I am not able to figure out how to annotate the Counter, I have tried several ways but it's still failing.